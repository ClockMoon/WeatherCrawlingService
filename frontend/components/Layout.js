import React, { useEffect } from "react";

const Layout = ({ children }) => {
  useEffect(() => {
    let totalPageSize;
    let fragment;
    let indexPage;
    let loginPage;
    let index;
    let login;
    window.addEventListener("load", function() {
      setTimeout(() => {
        totalPageSize = document.body.scrollHeight;
        fragment = totalPageSize / 5;
        indexPage = 1;
        loginPage = 0;
        index = fragment * indexPage;
        login = fragment * loginPage;
        window.scrollTo(0, index);
        console.log(totalPageSize);
        console.log(index);
      }, 100); //페이지 로드시에 즉시 body의 길이를 가져 올 때 조금씩 부족하게 나옴
    });

    window.addEventListener("scroll", function() {
      console.log(scrollY);
    });
  }, []);
  return (
    <div>
      <div className="header">
        <div className="title">
          <div className="colorFont">WeatherCrawler</div>
        </div>
        <div className="board">
          <div className="colorFont dashBoard">DashBoard</div>
          <div className="colorFont login">Login</div>
        </div>
      </div>
      <div>{children}</div>
    </div>
  );
};

export default Layout;
