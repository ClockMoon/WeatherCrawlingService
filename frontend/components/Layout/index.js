import React, { useEffect } from "react";
import { useSelector, useDispatch } from "react-redux";
import scroll from "scroll";
import scrollDoc from "scroll-doc";
import { LOAD_USER_REQUEST, loadUserAction } from "../../reducers/user";

const page = scrollDoc();
let fragment = 0;
let nowScrollLocation = fragment;

const Layout = ({ children }) => {
  const dispatch = useDispatch();
  const user = useSelector(state => state.user.user);
  useEffect(() => {
    let totalPageSize;
    let fragment;
    let indexPage;
    let loginPage;
    let index;
    let login;
    window.addEventListener("load", function() {
      setTimeout(() => {
        totalPageSize = document.body.scrollHeight;
        fragment = totalPageSize / 5;
        indexPage = 1;
        loginPage = 0;
        index = fragment * indexPage;
        login = fragment * loginPage;
        window.scrollTo(0, index);
      }, 100); //페이지 로드시에 즉시 body의 길이를 가져 올 때 조금씩 부족하게 나옴
    });
    window.addEventListener(
      "wheel",
      function(event) {
        event.preventDefault();
      },
      { passive: false }
    );
  }, []);

  const goDashBoard = e => {
    const totalPageSize = document.body.scrollHeight;
    fragment = totalPageSize / 5;
    nowScrollLocation = fragment;
    scroll.top(page, nowScrollLocation);
    const target = e.target;
    target.classList.add("activeJump");
    setTimeout(() => {
      target.classList.remove("activeJump");
    }, 1100);
  };

  const goLogin = e => {
    nowScrollLocation = 0;
    scroll.top(page, nowScrollLocation);
    const target = e.target;

    target.classList.add("activeJump");
    setTimeout(() => {
      target.classList.remove("activeJump");
    }, 1100);
  };

  return (
    <div>
      <div className="header">
        <div className="title">
          <div className="colorFont">WeatherCrawler</div>
        </div>
        <div className="board">
          <div onClick={goDashBoard} className="colorFont dashBoard">
            DashBoard
          </div>

          <div onClick={goLogin} className="colorFont login">
            {user ? "Info" : "Login"}
          </div>
        </div>
      </div>
      <div>{children}</div>
    </div>
  );
};
export default Layout;
