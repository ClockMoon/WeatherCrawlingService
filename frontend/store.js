import createSagaMiddleware from "redux-saga";
import rootSaga from "./sagas";
import reducer from "./reducers";
import { applyMiddleware, compose, createStore } from "redux";

export default (initialState, options) => {
  if (options) {
    const sagaMiddleware = createSagaMiddleware();
    const middlewares = [sagaMiddleware];
    const enhancer =
      process.env.NODE_ENV === "production"
        ? compose(applyMiddleware(...middlewares))
        : compose(
            applyMiddleware(...middlewares),
            !options.isServer &&
              typeof window.__REDUX_DEVTOOLS_EXTENSION__ !== "undefined"
              ? window.__REDUX_DEVTOOLS_EXTENSION__()
              : f => f
          );
    const store = createStore(reducer, initialState, enhancer);
    store.sagaTask = sagaMiddleware.run(rootSaga);
    return store;
  }
  const sagaMiddleware = createSagaMiddleware();
  const middlewares = [sagaMiddleware];
  const enhancer = compose(applyMiddleware(...middlewares));
  const store = createStore(reducer, initialState, enhancer);
  store.sagaTask = sagaMiddleware.run(rootSaga);
  return store;
};
