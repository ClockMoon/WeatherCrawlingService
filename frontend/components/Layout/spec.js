import { mount } from "enzyme";
import Layout from "./index";
import { Provider } from "react-redux";
import store from "../../store";
describe("Layout 컴포넌트 테스트.", () => {
  it("Layout의 header가 렌더링됩니다.", () => {
    const component = mount(
      <Provider store={store()}>
        <Layout />
      </Provider>
    );
    const header = component.find(".header");
    expect(header.debug()).toMatchSnapshot();
  });
});
