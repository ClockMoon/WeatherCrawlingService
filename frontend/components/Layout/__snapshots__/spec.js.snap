// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout 컴포넌트 테스트. Layout의 header가 렌더링됩니다. 1`] = `
"<div className=\\"header\\">
  <div className=\\"title\\">
    <div className=\\"colorFont\\">
      WeatherCrawler
    </div>
  </div>
  <div className=\\"board\\">
    <div onClick={[Function: goDashBoard]} className=\\"colorFont dashBoard\\">
      DashBoard
    </div>
    <div onClick={[Function: goLogin]} className=\\"colorFont login\\">
      Login
    </div>
  </div>
</div>"
`;
